neo/Ledger/MemoryPool.cs

```diff
@@ -347,25 +348,23 @@ internal void UpdatePoolForBlockPersisted(Block block, Snapshot snapshot)
             _txRwLock.EnterWriteLock();
             try
             {
+                // First remove the transactions verified in the block.
+                foreach (Transaction tx in block.Transactions)
+                {
+                    if (TryRemoveVerified(tx.Hash, out _)) continue;
+                    TryRemoveUnVerified(tx.Hash, out _);
+                }
+
+                // Add all the previously verified transactions back to the unverified transactions
+                InvalidateVerifiedTransactions();
+
                 if (policyChanged)
                 {
-                    _unsortedTransactions.Clear();
-                    _sortedTransactions.Clear();
+                    foreach (PoolItem item in _unverifiedSortedTransactions.Reverse())
+                        _system.Blockchain.Tell(item.Tx, ActorRefs.NoSender);
```

Comment:
Any chance that, in the time you retell, and then you clear already retelled tx? :thinking: